app/assets/javascripts/components.js

```diff
@@ -17,3 +17,9 @@ function cdx_select_on_change(name, callback) {
     callback($(this).val());
   });
 }
+$(document).on("change", "input[type=date]", function(){
+  $(this).val() != '' ? $(this).addClass("filled") : $(this).removeClass("filled");
+});
+$(function(){
+  $("input[type=date]").change();
+});
```

Comment:
When I see JS I'm always wondering if we could achieve the same effect in pure CSS, and quite often it happens that we can. Consider:

```scss
input[type=date]:not([value='']) {
  // style for filled element
}
```

Reply:
I'm also a little wary of this piece of JS, because it will affect _all_ date input types (same for the CSS selector above), and maybe we want it to to only apply to the date selector appearing in filters?

But maybe this case is fine, because we probably want all date inputs to look the same across the site :+1:

Reply:
(I also found the `:placeholder-shown` pseudo selector, but it doesn't match the default values of a date type, only inputs with an actual placeholder).

Reply:
It would be nice if `[value='']` worked. But the `value` attribute is not dynamically bound. It does not change when editing the field value in the UI. You could use a javascript handler to do that: `onchange="this.setAttribute('value', this.value);"`. But then we're back at JS 🤷 
An alternative would be `:not(:placeholder-shown)`, which requires a `placeholder` attribute (although it can be empty). Unfortunately, this does not work reliably with date inputs. Their widget typically doesn't show a placeholder.

Last option would be `:valid` with form validation attributes, but I think that's probably too much of a hack. Filled doesn't necessarily mean valid.

Reply:
@ysbaddaden the idea is exactly to make it work on every input date, because it's the default style of a input date, and it was wrong on other pages, like the initial dashboard.

I also found selectors like `:placeholder-shown`, `:valid`, `:empty`. I talked about this to @straight-shoota in somewhere, maybe another PR. They doesn't work. Input date works in a very particular way.

Reply:
I spent a long time before deciding to make it works like this. I know that it doesn't look good when you see it at the first time, and I think I should explain it better when I opened my PR, to help you with these doubts. My fault. That's the best way I found, because of many conditions.

Reply:
@ysbaddaden look what happens when I change the value of a input date, using the `[value=''"]` selector:
![image](https://user-images.githubusercontent.com/19227697/181765106-6fcc67b9-1503-496e-8c19-03430b7546c8.png)

Reply:
It's probably a good idea to add a short comment in the code to explain the reasoning for this solution. This will avoid future me and you from having to think about it again.
That's a good practice whenever you need to research a solution, to add a comment how you ended up with that. That makes it easy for a reviewer to understand your reasons and make reviews easier.

Reply:
Great idea @straight-shoota you are right. Even I can forget about it all. Should I put this on JS? I think it's the place where we have the most unexpected solution, right?